mempool/src/core_mempool/transaction_store.rs

```diff
@@ -443,6 +460,7 @@ impl TransactionStore {
         if let Some(txns) = self.transactions.get(address) {
             if txns.is_empty() {
                 self.transactions.remove(address);
+                self.sequence_numbers.remove(address);
```

Comment:
also why we don't  merge this function with index_remove?

Reply:
I folded the entire `remove_account_if_empty` into `index_remove`. One morehashmap lookup per txn, but that should be fine.